config/connection_settings.go

```diff
@@ -56,8 +56,13 @@ type AzureConnectionSettings struct {
 	*SharedAccessKey
 }
 
-// CreateAzureConnectionSettings creates the configuration data for establishing connection to the Azure IoT Hub.
+// CreateAzureConnectionSettings creates the configuration data for establishing connection to the Azure IoT Hub, using no IDScopeProvider.
 func CreateAzureConnectionSettings(settings *AzureSettings, log logger.Logger) (*AzureConnectionSettings, error) {
+	return PrepareAzureConnectionSettings(settings, nil, log)
+}
+
+// PrepareAzureConnectionSettings creates the configuration data for establishing connection to the Azure IoT Hub, allowing usage of IDScopeProvider.
+func PrepareAzureConnectionSettings(settings *AzureSettings, idScopeProvider IDScopeProvider, log logger.Logger) (*AzureConnectionSettings, error) {
```

Comment:
If we keep Create, the name should be revised.

Reply:
See my comment from above, Create is the original method and it is not really needed.